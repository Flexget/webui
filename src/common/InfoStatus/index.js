import { connect } from 'react-redux';
import StatusBar from 'common/StatusBar';
import actions from 'core/status/state/actions';

export function mapStateToProps({ status }) {
  return {
    open: !!status.info,
    message: status.info,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(actions.clear()),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(StatusBar);
