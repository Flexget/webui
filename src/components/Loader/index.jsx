import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';

const Loader = ({
  isLoading,
  timedOut,
  pastDelay,
  error,
}) => {
  if (isLoading) {
    // While our other component is loading...
    if (timedOut) {
      // In case we've timed out loading our other component.
      return <div>Loader timed out!</div>;
    } if (pastDelay) {
      // Display a loading screen after a set delay.
      return <CircularProgress />;
    }
    // Don't flash "Loading..." when we don't need to.
    return null;
  }

  if (error) {
    // If we aren't loading, maybe
    return <div>Error! Component failed to load</div>;
  }
  // This case shouldn't happen... but we'll return null anyways.
  return null;
};

Loader.propTypes = {
  isLoading: PropTypes.bool,
  timedOut: PropTypes.bool,
  pastDelay: PropTypes.bool,
  error: PropTypes.object, // eslint-disable-line react/require-default-props
};

Loader.defaultProps = {
  isLoading: false,
  timedOut: false,
  pastDelay: false,
};

export default Loader;
