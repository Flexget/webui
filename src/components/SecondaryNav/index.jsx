import React from 'react';
import PropTypes from 'prop-types';
import { MuiThemeProvider } from 'material-ui/styles';
import { darkTheme } from 'theme';
import {
  SecondaryAppBar,
  SecondaryToolbar,
  SecondaryTabs,
} from './styles';

const SecondaryNav = ({ children, tabs, ...tabProps }) => (
  <MuiThemeProvider theme={darkTheme}>
    <SecondaryAppBar>
      { tabs ?
        (
          <SecondaryTabs
            indicatorColor="primary"
            {...tabProps}
          >
            {children}
          </SecondaryTabs>
        ) :
        (
          <SecondaryToolbar>
            {children}
          </SecondaryToolbar>
        )
      }
    </SecondaryAppBar>
  </MuiThemeProvider>
);

SecondaryNav.propTypes = {
  children: PropTypes.node.isRequired,
  tabs: PropTypes.bool,
};

SecondaryNav.defaultProps = {
  tabs: false,
};

export default SecondaryNav;
