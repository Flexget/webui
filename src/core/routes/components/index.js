import { connect } from 'react-redux';
import Routes from './Routes';

function mapStateToProps({ routes }) {
  return Object.values(routes);
}

export default connect(mapStateToProps)(Routes);
