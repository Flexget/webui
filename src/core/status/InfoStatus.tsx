import React, { FC } from 'react';
import StatusBar from './StatusBar';
import { actions, StatusContainer } from './hooks';

const InfoStatus: FC = () => {
  const [{ info }, dispatch] = StatusContainer.useContainer();
  const clearStatus = () => {
    dispatch(actions.popInfo());
  };

  return <StatusBar message={info[0]} clearStatus={clearStatus} />;
};

export default InfoStatus;
