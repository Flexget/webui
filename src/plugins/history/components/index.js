import { connect } from 'react-redux';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'plugins/history/data/actions';
import History from './History';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(History);
export { History };
