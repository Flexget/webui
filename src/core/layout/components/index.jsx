import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'core/layout/components/Logo';
import Navbar from 'core/layout/components/Navbar';
import SideNav from 'core/layout/components/SideNav';
import LoadingBar from 'components/LoadingBar';
import ErrorStatus from 'components/ErrorStatus';
import InfoStatus from 'components/InfoStatus';
import {
  Wrapper,
  Main,
  Header,
  LogoWrapper,
  Nav,
  SideBar,
  Content,
} from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState(({ sideBarOpen }) => ({
      sideBarOpen: !sideBarOpen,
    }));
  }

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav
              sideBarOpen={sideBarOpen}
              toggle={this.toggleSideBar}
            />
          </SideBar>
          <Content open={sideBarOpen}>
            { children }
          </Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
