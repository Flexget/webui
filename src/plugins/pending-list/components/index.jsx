import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { NoPaddingWrapper } from 'components/styles';
import TabList from './TabList';

class PendingList extends PureComponent {
  state = {
    listId: false,
  }

  selectList = listId => this.setState({ listId })

  render() {
    const { listId } = this.state;

    return (
      <NoPaddingWrapper>
        <TabList
          listId={listId}
          selectList={this.selectList}
        />
      </NoPaddingWrapper>
    );
  }
}

export default hot(module)(PendingList);
