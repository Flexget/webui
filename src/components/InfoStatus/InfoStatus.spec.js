import { mapStateToProps } from 'components/InfoStatus';

describe('components/InfoStatus', () => {
  it('should be correct if an info status should be displayed', () => {
    expect(mapStateToProps({ status: { info: 'Info Status' } })).toMatchSnapshot();
  });

  it('should be correct if an info status should not be displayed', () => {
    expect(mapStateToProps({ status: { } })).toMatchSnapshot();
  });
});
