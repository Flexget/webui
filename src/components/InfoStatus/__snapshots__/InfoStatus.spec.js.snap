// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/components/InfoStatus should be correct if an info status should be displayed 1`] = `
Object {
  "message": "Info Status",
  "open": true,
}
`;

exports[`plugins/components/InfoStatus should be correct if an info status should not be displayed 1`] = `
Object {
  "message": undefined,
  "open": false,
}
`;
